src/core/adaptersController.ts

```diff
@@ -95,54 +93,81 @@ export class AdaptersController {
   private async buildProtocolTokens(): Promise<
     Map<Chain, Map<string, IProtocolAdapter>>
   > {
-    const protocolTokens: Map<Chain, Map<string, IProtocolAdapter>> = new Map()
+    const protocolTokensAdapterMap: Map<
+      Chain,
+      Map<string, IProtocolAdapter>
+    > = new Map()
 
     for (const [chainId, chainAdapters] of this.adapters) {
-      protocolTokens.set(chainId, new Map())
-      const chainAdaptersMap = protocolTokens.get(chainId)!
+      protocolTokensAdapterMap.set(chainId, new Map())
+      const chainAdaptersMap = protocolTokensAdapterMap.get(chainId)!
 
       for (const [_protocolId, protocolAdapters] of chainAdapters) {
         for (const [_productId, adapter] of protocolAdapters) {
           const { positionType } = adapter.getProtocolDetails()
 
-          if (positionType === PositionType.Reward) {
-            continue
-          }
-
           let protocolTokens: Erc20Metadata[]
           try {
             protocolTokens = await adapter.getProtocolTokens()
           } catch (error) {
             if (!(error instanceof NotImplementedError)) {
               throw error
             }
-            protocolTokens = []
+            continue
           }
-          for (const protocolToken of protocolTokens) {
-            const tokenAddress = protocolToken.address.toLowerCase()
 
-            if (_protocolId === 'prices') {
-              if (chainAdaptersMap.has(tokenAddress)) {
-                continue
-              }
-            } else {
-              if (chainAdaptersMap.has(tokenAddress)) {
-                const existingAdapter = chainAdaptersMap.get(tokenAddress)
-                if (existingAdapter?.protocolId !== 'prices') {
-                  throw new Error(
-                    `Duplicated protocol token ${protocolToken.address}`,
-                  )
-                }
-              }
-            }
-
-            chainAdaptersMap.set(tokenAddress, adapter)
+          switch (positionType) {
+            case PositionType.FiatPrices:
+              this.processFiatPrices(protocolTokens, chainAdaptersMap, adapter)
+              break
+            case PositionType.Reward:
+              // Handle Reward case here if needed
+              break
+            default:
+              this.processDefaultCase(protocolTokens, chainAdaptersMap, adapter)
+              break
```

Comment:
Not related to this feature, this is an additional refactor to improve readability